eval.c

```diff
@@ -1232,10 +1232,7 @@ ensure_class_or_module(VALUE obj)
 static VALUE
 hidden_identity_hash_new(void)
 {
-    VALUE hash = rb_ident_hash_new();
-
-    RBASIC_CLEAR_CLASS(hash); /* hide from ObjectSpace */
```

Comment:
Looking through uses of `rb_ident_hash_new`, I ran into this. Most cases use `rb_obj_hide`, but this uses `RBASIC_CLEAR_CLASS`. Is the difference intentional?

Reply:
@nobu Do you have any context on this?

Reply:
This line predates commit:adf1c94ffe75, where `rb_obj_hide` was added.
Just added to fix https://bugs.ruby-lang.org/issues/8204, but didn't change already existing code all.
Maybe for unnecessary `SPECIAL_CONST_P` check.
It would be OK to replace this line with `rb_obj_hide`, I think.